core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java

```diff
@@ -281,4 +283,18 @@ public FlowIdentifier updatePillarConfiguration(NameOrCrn nameOrCrn, Long worksp
         }
         return clusterOperationService.updatePillarConfiguration(stack);
     }
+
+    public CertificatesRotationV4Response rotateClusterCertificates(NameOrCrn nameOrCrn, Long workspaceId,
+            CertificatesRotationV4Request certificatesRotationV4Request) {
+        Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
+        MDCBuilder.buildMdcContext(stack);
+        if (!stack.isAvailable()) {
+            throw new BadRequestException(String.format(
+                    "Stack '%s' is currently in '%s' state. Certificates rotation can "
+                            + "only be made when the underlying stack is 'AVAILABLE'.",
+                    stack.getName(),
+                    stack.getStatus()));
+        }
```

Comment:
this is a duplicate except a few words in the message. please consider refactoring it into a common method